### DISCUSSION TITLE
Native Support for Alternate Geospatial CRS (EPSG:4326)?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#6216] [Q&A] [answered]
Hi! 

Over the past ~year I have made multiple unsuccessful  attempts ([5607](https://github.com/visgl/deck.gl/discussions/5607) , [6090](https://github.com/visgl/deck.gl/discussions/6090)) at trying to leverage CRS EPSG:4326 tile sources with deck in a geospatial capacity.

Was curious what the process is to nominate a feature request to prioritize adding native support for alternate geospatial CRS’ other than EPSG:3857? Namely adding support to MapController/MapView to support EPSG:4326?

When working on attempt [6090](https://github.com/visgl/deck.gl/discussions/6090) @Pessimistress made mention of this RFC (https://github.com/visgl/deck.gl/blob/master/dev-docs/RFCs/proposals/projection-mode-improvements-rfc.md) from 2017.

When working on attempt [5607](https://github.com/visgl/deck.gl/discussions/5607) @kylebarron mentioned this proposed draft (https://github.com/visgl/deck.gl/pull/5504) as a possibility.

My current solution is less than ideal as I have stood up a MapProxy server that reprojects tiles from EPSG:4326 to EPSG:3857.

Would love to be able to leverage decks MapController/MapView without having to do this reprojection via MapProxy.

Others appear to have expressed similar interest…

https://github.com/visgl/deck.gl/discussions/6058

Thanks so much for your consideration and all you are doing creating such an awesome framework.

**Comment by Pessimistress** [comment, marked as answer]
We have a large feature backlog. I am personally maintaining this project on my own time, so I cannot promise an ETA for this particular feature. Here are things you can do:

- Create a feature request and gather interest from other users. We always try work on features that can benefit the most users, though again, no guaranteed timeline.
- If you are willing to implement this feature yourself, I'm happy to support you either in a discussion thread or the [Slack channel](https://github.com/visgl/deck.gl#learning-resources). I apologize if I appear to be unresponsive to #6090; it seems that you need me to write a significant amount of code for you to proceed, which comes back to the resource issue.
- If you need prioritized technical support for your use case, you or your organization can become a sponsor of the [Urban Computing Foundation](https://uc.foundation/).

**Reply by jfuehner** [reply]
@Pessimistress thanks for your response. I don't think I realized you were personally supporting deck on your own time. Its a wonderful project and I am super appreciative of the work you are doing.

I created FR #6217

**Comment by BranZhang** [comment]
I did some work to make Mapbox and deck.gl support 4326 projection.

online demo, maybe a little slow, because some js files are deployed on Chinese server.
https://jsfiddle.net/little_potato/ocsf4bwt/

I modified Mapbox to support 4326 projection.
```javascript
aimap.srs = 'epsg:4326';
```

And I modified deck.gl base on v8.6.5 to support 4326 projection. this file:
```html
<script src="https://location-dev.newayz.com/aimap-gl-js/libs/deck-4326.min.js"></script>
```

mostly change functions about projection between web-mercator and lnglat.

![image](https://user-images.githubusercontent.com/10242480/148040292-893d76c6-9d35-42be-8a51-ebad76706dda.png)

**Reply by Baruch-G** [reply]
Thanks for sharing!
Any chance you have those Deck.gl modifications based on later versions ?

**Reply by amuedespacher** [reply]
Hi @BranZhang , any chance you could share the source code of your adaptations to make deck.gl render in a projection other than web mercator?

**Reply by BranZhang** [reply]
@Baruch-G , @amuedespacher , my pleasure, I will share my source code and white a blog about 4326 projection. wait me about 1~2 week.

**Reply by BranZhang** [reply]
@Baruch-G , @amuedespacher , 😊my code: https://github.com/BranZhang/deck.gl

**Reply by amuedespacher** [reply]
Thank you, @BranZhang ; I'll have a look at it

**Reply by ovadia-nisimov** [reply]
Hi @BranZhang ,
I'm trying to use a WMTS tiles map in EPSG:4326 with deck.gl.
Has there been any progress since your last update on this project?
I would appreciate it if you could share your solution.

**Reply by BranZhang** [reply]
@ovadia-nisimov , no update. maybe you can try using [MapProxy](https://github.com/mapproxy/mapproxy) to translate 4326 WMTS to 3857.

**Comment by BranZhang** [comment]
When I modified deck.gl for supporting 4326 projection, I found this code in 'viewport.js':
```javascript
get projectionMode() {
    if (this.isGeospatial) {
        // I modified
        return PROJECTION_MODE.WEB_MERCATOR;
        // deck.gl source code
        // return this.zoom < 12
        //   ? PROJECTION_MODE.WEB_MERCATOR
        //   : PROJECTION_MODE.WEB_MERCATOR_AUTO_OFFSET;
    }
    return PROJECTION_MODE.IDENTITY;
}
```
@Pessimistress I don't understand why projection mode changed to mercator-auto-offset when the map zoom level bigger than 12?
looks like for a higher display precision?

**Reply by ibgreen** [reply]
https://medium.com/vis-gl/how-sometimes-assuming-the-earth-is-flat-helps-speed-up-rendering-in-deck-gl-c43b72fd6db4

**Reply by BranZhang** [reply]
@ibgreen , thanks, very helpful.

**Comment by ovadia-nisimov** [comment]
Thanks

בתאריך יום ד׳, 25 בפבר׳ 2026 ב-8:16 מאת BranZhang ***@***.***
